cardano-node/src/Cardano/Tracing/OrphanInstances/Network.hs

```diff
@@ -1735,10 +1736,35 @@ instance Show vNumber => ToJSON (HandshakeException vNumber) where
                  ]
 
 instance ToJSON NodeToNodeVersion where
-  toJSON x = String (pack $ show x)
+  toJSON NodeToNodeV_7  = Number 7
+  toJSON NodeToNodeV_8  = Number 8
+  toJSON NodeToNodeV_9  = Number 9
+  toJSON NodeToNodeV_10 = Number 10
+
+instance FromJSON NodeToNodeVersion where
+  parseJSON (Number 7) = return NodeToNodeV_7
+  parseJSON (Number 8) = return NodeToNodeV_8
+  parseJSON (Number 9) = return NodeToNodeV_9
+  parseJSON (Number 10) = return NodeToNodeV_10
+  parseJSON (Number x) = fail ("unsupported node-to-node protocol version " ++ show x)
+  parseJSON _          = fail "error parsing NodeToNodeVersion"
 
 instance ToJSON NodeToClientVersion where
-  toJSON x = String (pack $ show x)
+  toJSON NodeToClientV_9  = Number 9
+  toJSON NodeToClientV_10 = Number 10
+  toJSON NodeToClientV_11 = Number 11
+  toJSON NodeToClientV_12 = Number 12
+  toJSON NodeToClientV_13 = Number 13
+  toJSON NodeToClientV_14 = Number 14
+
+instance FromJSON NodeToClientVersion where
+  parseJSON (Number 9) = return NodeToClientV_9
+  parseJSON (Number 10) = return NodeToClientV_10
+  parseJSON (Number 11) = return NodeToClientV_11
+  parseJSON (Number 12) = return NodeToClientV_12
+  parseJSON (Number 13) = return NodeToClientV_13
+  parseJSON (Number x) = fail ("unsupported node-to-client protocol version " ++ show x)
+  parseJSON _          = fail "error parsing NodeToClientVersion"
```

Comment:
This would be better `"FromJSON.NodeToClientVersion: expected Number but got " <> show aesonValue`